### DISCUSSION TITLE
Python implementation of "Plot Spectrum"

### DISCUSSION THREAD

**Discussion by AishwaryaMukund** [audacity/audacity#1262] [Q&A] [answered]
Hi,

I tried to call the "Plot Spectrum" function from a Python script using the script `pipe_test.py` and https://manual.audacityteam.org/man/scripting_reference.html#Extra:_Scriptables_II

```  
    do_command('Import2: Filename="chunk1.wav"') 
    do_command('Select: Start=46 End=49') 
    do_command('PlotSpectrum: ')
 ```

Now I want to export the x & y axis values of resulting graph (db vs frequency). Is there a way this can be done? 

Or if you have any ideas on other Python implementation  for generating db vs frequency graphs, please do share.

**Comment by SteveDaulton** [comment, marked as answer]
Perhaps with pysox.

Example: print the frequency of the highest peak in the power spectrum, where "sine.wav" is a WAV file in the current working directory.
(It's not very fast, so ensure that the input file is quite short, at least while testing).

```
#! /usr/bin/python3
# -*- coding: utf-8 -*-

import os
import sox

wav_path = os.path.join(os.getcwd(), "sine.wav")
tfm = sox.Transformer()

peak = [0, 0]
spectrum = tfm.power_spectrum(wav_path)

for pair in spectrum:
    if pair[1] > peak[1]:
        peak = pair

print(f'Peak frequency is {peak[0]} Hz')

```
